### DISCUSSION TITLE
Any plan for command completion ?

### DISCUSSION THREAD

**Discussion by xaviermignot** [Azure/azure-dev#125] [General]
Pretty much what title says, is there any plan to have command completion in terminal ?
Just like what we have with Azure CLI, autocompletion works great in my zsh shell 🤓 

Many thanks in advance 🙏

**Comment by savannahostrowski** [comment]
Are there any commands that you finding tricky to remember or particularly lengthy? Our commands are intentionally quite short so I'd be interested to know if there are particular instances where autocomplete would be helpful.

**Reply by xaviermignot** [reply]
It's not a matter of command length, it's more on the experience I'm used to when I use my terminal, even for simple CLIs I find myself using the completion quite a lot.
I agree on the shortness of the azd commands, but as I am not using the tool everyday (for now at least), I prefer to use completion instead of the `-h` flag all the time when I don't remember something :)

**Comment by ellismg** [comment]
This should be easy-ish for us to add - cobra (the library we use for our CLI) supports generating completions.

**Comment by savannahostrowski** [comment]
In case anyone comes looking here, completions were added!
https://github.com/Azure/azure-dev#set-up-shell-completion

**Reply by xaviermignot** [reply]
I was still looking here, I have just set it up and I am loving it !

The command `azd completion zsh --help` did not return any instruction on how to set-up completion though (it return the same content as `azd --help`).
I have managed to get it working using [this answer](https://stackoverflow.com/a/66145365) from SO.

Many thanks again :)

**Comment by ThatRendle** [comment]
For anyone finding this looking for Powershell setup, add this line to your `$PROFILE`:

```powershell
Invoke-Expression (& { (azd completion powershell | Out-String) })
```
